contracts/middle-layer/GreenfieldExecutor.sol

```diff
@@ -0,0 +1,67 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "../Config.sol";
+import "../interface/ICrossChain.sol";
+import "../interface/IGreenfieldExecutor.sol";
+
+contract GreenfieldExecutor is Config, Initializable, IGreenfieldExecutor {
+    // Supported message types and its corresponding number
+    // 1: CreatePaymentAccount
+    // 2: Deposit
+    // 3: DisableRefund
+    // 4: Withdraw
+    // 5: MigrateBucket
+    // 6: CancelMigrateBucket
+    // 7: CompleteMigrateBucket
```

Comment:
`CompleteMigrateBucket` and `RejectMigrateBucket` seems not send by a user, but a SP.